### PR TITLE
fix(db): eliminate uncategorized skills via dedup migration and pivot sync fix

### PR DESCRIPTION
**Pull request by ericchansen**
## Summary

Eliminates the 111 uncategorized skills in production by fixing the root causes: pivot sync creating skills without categories, Excel dedup suffixes creating duplicates, and stale seed data.

## Changes

| File | Change |
|------|--------|
| `backend/data/skill-name-map.json` | Canonical name aliases and Soft Skills list |
| `backend/utils/normalizeSkill.js` | Strip Excel dedup suffixes (e.g., `Azure Container Apps2` → `Azure Container Apps`) |
| `backend/migrate.js` | Idempotent cleanup migration: merge dupes, assign categories, add unique index |
| `backend/services/sharepoint.js` | Fix `syncPivotToDatabase()` to normalize names and assign categories; MAX wins for dedup ratings |
| `scripts/generate_seed.py` | `ON CONFLICT DO UPDATE` instead of `DO NOTHING`; add Soft Skills category |
| `database/seed.sql` | Regenerated: 55 categories (was 54), 153 skills (was 145) |

## Root Cause

1. `syncPivotToDatabase()` created skills with `NULL category_id` when pivot column names didn't match existing skills
2. Excel pivot dedup suffixes (`Azure Functions3`) created separate uncategorized skills
3. `seed.sql` used `ON CONFLICT DO NOTHING` so re-running init couldn't fix stale data

## Migration Details

On next backend startup, `migrate.js` will:
1. Create "Soft Skills" top-level category with 8 SE skills (presentations, whiteboarding, etc.)
2. Merge duplicate skills by moving `user_skills` references (MAX proficiency wins) and deleting dupes
3. Assign categories to remaining uncategorized skills via name matching
4. Add `UNIQUE INDEX` on `skills.name` to prevent future duplicates

## Testing

- 101/101 backend tests passing
- `normalizeSkillName()` verified for all known edge cases (SQL DB, L200, Power BI, etc.)
- Frontend builds cleanly

### CODE REVIEW

File: backend/services/sharepoint.js [hunk 1]
```diff
@@ -272,20 +273,31 @@ async function syncPivotToDatabase(pivotData) {
...
-  // Phase 1: Upsert all skills from column headers
+  // Phase 1: Upsert all skills from column headers (with name normalization)
   const skillIdMap = new Map();
-  for (const skillName of skillNames) {
+  for (const rawName of skillNames) {
+    const skillName = normalizeSkillName(rawName);
+
+    // If we already resolved this canonical name, reuse the same ID
+    if (skillIdMap.has(rawName)) continue;
+
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply backend/services/sharepoint.js]: Fixed in ad93a23. Added canonicalIdMap keyed by normalized name so multiple raw names that resolve to the same canonical skill share a single DB lookup.
